kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/LikeExpressionEvaluator.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.expressions;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.expressions.Expression;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.expressions.Predicate;
+import io.delta.kernel.types.BooleanType;
+import io.delta.kernel.types.DataType;
+import io.delta.kernel.types.StringType;
+import io.delta.kernel.internal.util.Utils;
+
+import static io.delta.kernel.defaults.internal.DefaultEngineErrors.invalidEscapeSequence;
+import static io.delta.kernel.defaults.internal.DefaultEngineErrors.unsupportedExpressionException;
+
+/**
+ *
+ */
+public class LikeExpressionEvaluator {
+    private LikeExpressionEvaluator() {
+    }
+
+    static Predicate validateAndTransform(
+            Predicate like,
+            Expression left, DataType leftOutputType,
+            Expression right, DataType rightOutputType,
+            Expression escapeCharExpr, DataType escapeCharOutputType) {
+
+        if (!(StringType.STRING.equivalent(leftOutputType)
+                && StringType.STRING.equivalent(rightOutputType))) {
+            throw unsupportedExpressionException(like,
+                    "'like' is only supported for string type expressions");
```

Comment:
nit: lets use `LIKE` to emphasize the expression name and remove the quotes around it.